test/inference.jl

```diff
@@ -843,3 +843,12 @@ f21771(::Val{U}) where {U} = Tuple{g21771(U)}
 # ensure that we don't try to resolve cycles using uncached edges
 f21653() = f21653()
 @test code_typed(f21653, Tuple{}, optimize=false)[1] isa Pair{CodeInfo, typeof(Union{})}
+
+# constant inference of isdefined
+let f(x) = isdefined(x,2) ? 1 : ""
+    @test Base.return_types(f, (Tuple{Int,Int},)) == Any[Int]
+    @test Base.return_types(f, (Tuple{Int,},)) == Any[String]
+end
+let f(x) = isdefined(x,:re) ? 1 : ""
+    @test Base.return_types(f, (Complex64,)) == Any[Int]
+end
```

Comment:
need test coverage for:
 - isdefined with a symbol that isn't a valid field name
 - isdefined with a non-symbol-or-integer
 - isdefined with `x` is an: `SimpleVector`, `Module`, `Type`, `Array`, `Symbol`
 - isdefined on a type where the field is not necessarily assigned
 - isdefined on an abstract type
 - isdefined on a UnionAll type